plasmapy/physics/parameters.py

```diff
@@ -524,11 +520,177 @@ def kappa_thermal_speed(T, kappa, particle="e", method="most_probable"):
+        The number density of the Maxwellian test electrons
+
+    ion_particle: string
+        String signifying a particle type of the field ions, including charge
+        state information.
+
+    coulomb_log : float or dimensionless ~astropy.units.Quantity, optional
+        option to specify a Coulomb logarithm of the electrons on the ions.
+        If not specified, the Coulomb log will is calculated using the
+        .transport/Coulomb_logarithm function.
+
+    References
+    ----------
+    .. [1] Braginskii
+
+    .. [2] Formulary
+
+    """
+    from plasmapy.physics.transport import Coulomb_logarithm
+    T_e = T_e.to(units.K, equivalencies=units.temperature_energy())
+    if coulomb_log is not None:
+        coulomb_log_val = coulomb_log
+    else:
+        particles = ['e', ion_particle]
+        coulomb_log_val = Coulomb_logarithm(T_e, n_e, particles, V)
+    Z_i = atomic.charge_state(ion_particle)
+    nu_e = 4 / 3 * np.sqrt(2 * np.pi / m_e) / (4 * np.pi * eps0)**2 * e**4 * \
+        n_e * Z_i * coulomb_log_val / (k_B * T_e)**1.5
+    return nu_e.to(1 / units.s)
+
+
+@utils.check_quantity({
+    'T_i': {'units': units.K, 'can_be_negative': False},
+    'n_i': {'units': units.m**-3, 'can_be_negative': False}
+})
+def ion_ion_collision_rate(T_i, n_i, ion_particle,
+                           coulomb_log=None, V=None):
+    r"""
+    momentum relaxation ion-ion collision rate
+
+    From Callen Chapter 2, http://homepages.cae.wisc.edu/~callen/chap2.pdf,
+    equations (2.17) and (2.120)
+
+    Considering a Maxwellian distribution of "test" ions colliding with
+    a Maxwellian distribution of "field" ions.
+
+    Note, it is assumed that electrons are present in such numbers as to
+    establish quasineutrality, but the effects of the test ions colliding
+    with them are not considered here.
+
+    This result is an ion momentum relaxation rate, and is used in many
+    classical transport expressions. It is equivalent to:
+    * 1/tau_i from ref [1]_ eqn (1) pp. #,
+    * 1/tau_i from ref [2]_ eqn (1) pp. #,
+    * nu_i\i_S from ref [2]_ eqn (1) pp. #,
+
+    Parameters
+    ----------
+
+    T_i : ~astropy.units.Quantity
+        The electron temperature of the Maxwellian test ions
+
+    n_i : ~astropy.units.Quantity
+        The number density of the Maxwellian test ions
+
+    ion_particle: string
+        String signifying a particle type of the test and field ions,
+        including charge state information. This function assumes the test
+        and field ions are the same species.
+
+    coulomb_log : float or dimensionless ~astropy.units.Quantity, optional
+        option to specify a Coulomb logarithm of the electrons on the ions.
+        If not specified, the Coulomb log will is calculated using the
+        .transport/Coulomb_logarithm function.
+
+    References
+    ----------
+    .. [1] Braginskii
+
+    .. [2] Formulary
+
+    """
+    from plasmapy.physics.transport import Coulomb_logarithm
```

Comment:
Hmmm, should this be at top of file to be more consistent with the rest of plasmapy? Or are you trying to do some sort of optimization here?

Reply:
Well, the weird thing here is that at the top of `transport.py` line, you'll find

    from .parameters import (Debye_length, Hall_parameter,
                             electron_ion_collision_rate, ion_ion_collision_rate)

So we either do this in the function or we have a circular import. The latter of which is bad for your functions being able to get imported and executed.

I really do think `physics` is getting too bloated... but for now, this seems like the least bad solution.

Reply:
So does this indeed avoid the circular dependency problem? If so, that's kind of a neat hack.

Reply:
Actually nevermind, I straight up have no idea how the hell this is working - or if it's even working at all. >_<

Reply:
Haha, yeah, I'll admit this was a hack, as I recall, I ran into the circular imports thing, googled, saw it mentioned on stackoverflow that you could sometimes get away with this, tried it and it seemed to work. I don't understand what's really going on with the imports.